scripts/javasc.js

```diff
@@ -26,21 +26,118 @@ function userFeedback(userResponse, correctAnswer){
         return 0;
     }
 }
+ 
+function iterateQuestions(questions, answers, acceptedAnswers){
+    // Main loop to cycle through the questions
+    for (let i = 0; i < questions.length; i++) {
+        let ans = prompt(questions[i]);
+        while (!inputChecker(ans, acceptedAnswers)) {
+            alert('Please provide a valid input y/n/yes/no');
+            ans = prompt(questions[i]);
+        }
+
+        numberOfCorrect += userFeedback(ans, answers[i]);
+    } 
+}

```

Comment:
I really like this function! Very cool that it iterates through all your questions, validates responses, and tracks score. However, it's good to get into good habits while we're learning, so I'm going to mention the Single Responsibility Principle, or SRP for short. The short of that is that each function should only have a single responsibility, be that looping through the questions and checking answers, or validating responses. Utilizing the SRP will make your code more readable, as well as help with bug fixing.

I also really like that despite the fact that you're only ever going to plug one set of questions, answers, and acceptedAnswers into this function, they're all arguments. Scalable code is a beautiful thing.